src/sc_keeper/rate_limit.py

```
@@ -0,0 +1,271 @@
+        self.windows[key].append((now, credit_cost))
+        remaining = limit - (total_credits + credit_cost)
+        return True, remaining
+
+
+class RedisRateLimiter(RateLimiter):
+    """Redis-based rate limiter using sliding window with credit-based tracking."""
+
+    def __init__(self, redis_url: str, credits_per_minute: int):
+        redis_client = get_redis_client(redis_url)
+        if redis_client is None:
+            raise ImportError("Could not connect to Redis")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)